src/components/TagSet/TagSet.test.tsx

```diff
@@ -0,0 +1,335 @@
+    );
+  });
+
+  it("logs a warning when the `onClick` prop is passed", () => {
+    const warn = jest.spyOn(console, "warn");
+    // We cannot pass the `onClick` prop when the `type` is "explore",
+    // but because this is a test, we want to verify the logged message.
+    render(
+      // @ts-ignore
+      <TagSet
+        onClick={() => {}}
+        tagSetData={exploreTagSetData.simple}
+        type="explore"
+      />
+    );
+    expect(warn).toHaveBeenCalledWith(
+      "NYPL Reservoir TagSet: The `onClick` prop will be ignored when the `type` prop is set to 'explore'."
+    );
+  });
+
+  it("logs a warning when the labels are strings", () => {
+    const warn = jest.spyOn(console, "warn");
+    // We cannot pass labels as strings when the `type` is "explore", but because
+    // this is a test, we want to verify the logged message.
+    // @ts-ignore
+    render(<TagSet tagSetData={filterTagSetData.simple} type="explore" />);
+    expect(warn).toHaveBeenCalledWith(
+      "NYPL Reservoir TagSet: Explore tags require all `label` props to be React components."
+    );
+  });
+
+  it("renders the UI snapshot correctly", () => {
+    const simple = renderer
+      .create(<TagSet tagSetData={exploreTagSetData.simple} type="explore" />)
+      .toJSON();
+    const withIcons = renderer
+      .create(<TagSet tagSetData={exploreTagSetData.withIcon} type="explore" />)
+      .toJSON();
+    const withChakraProps = renderer
+      .create(
+        <TagSet
+          p="s"
+          color="ui.error.primary"
+          tagSetData={exploreTagSetData.simple}
+          type="explore"
+        />
+      )
+      .toJSON();
+    const withOtherProps = renderer
+      .create(
+        <TagSet
+          data-testid="testid"
+          tagSetData={exploreTagSetData.simple}
+          type="explore"
+        />
+      )
+      .toJSON();
+
+    expect(simple).toMatchSnapshot();
+    expect(withIcons).toMatchSnapshot();
+    expect(withChakraProps).toMatchSnapshot();
+    expect(withOtherProps).toMatchSnapshot();
+  });
+});
+
+describe("TagSet Filter", () => {
+  let onClick: jest.MockedFunction<() => void>;
+  // let utils: RenderResult;
+
+  it("renders tags", () => {
+    onClick = jest.fn();
+    // Seven color tags were passed as tags to display.
+    render(
+      <TagSet
+        onClick={onClick}
+        tagSetData={filterTagSetData.simple}
+        type="filter"
+      />
+    );
+
+    expect(screen.getAllByRole("button")).toHaveLength(7);
+  });
+
+  it("renders icons within the tags", () => {
+    onClick = jest.fn();
+    // Seven icon names were passed to display.
+    render(
+      <TagSet
+        onClick={onClick}
+        tagSetData={filterTagSetData.withIcon}
+        type="filter"
+      />
+    );
+
+    expect(screen.getAllByTestId("ts-icon")).toHaveLength(7);
+  });
+
+  it("renders close icons when `isDismissible` is true", () => {
```

Comment:
Should there be a warning that the icons will not render if isDismissable is true?

Reply:
Added.